test/MerkleLib.utils.ts

```diff
@@ -54,12 +54,16 @@ export function buildRelayerRefundLeaves(
 }
 
 export async function buildPoolRebalanceLeafTree(poolRebalanceLeaves: PoolRebalanceLeaf[]) {
-  for (let i = 0; i < poolRebalanceLeaves.length; i++) {
-    // The 4 provided parallel arrays must be of equal length.
-    expect(poolRebalanceLeaves[i].l1Tokens.length)
-      .to.equal(poolRebalanceLeaves[i].bundleLpFees.length)
-      .to.equal(poolRebalanceLeaves[i].netSendAmounts.length)
-      .to.equal(poolRebalanceLeaves[i].runningBalances.length);
+  for (const leaf of poolRebalanceLeaves) {
+    const { l1Tokens, bundleLpFees, netSendAmounts, runningBalances } = leaf;
+
+    // l1Tokens, bundleLpFees and netSendAmounts must always be of equal length.
+    expect(l1Tokens.length).to.equal(bundleLpFees.length).to.equal(netSendAmounts.length);
```

Comment:
didn't know you could do this. Have you tested that this double `.equal()` actually works and doesn't return early after the first `.equal()` returns true?

Reply:
> didn't know you could do this. Have you tested that this double `.equal()` actually works and doesn't return early after the first `.equal()` returns true?

This is basically the same logic as was there before, so I assumed it worked. I'll verify it now.

Reply:
@nicholaspai Looks like it fails as expected, with the following test:

```
 expect(0).to.equal(0).to.equal(1);
```

![2023-06-06-160923_1791x504_scrot](https://github.com/across-protocol/contracts-v2/assets/108695806/7bc69619-04a2-4c56-8b04-c829e470df2f)